src/test/java/teammates/test/cases/action/InstructorCourseDeleteRecoveryCourseActionTest.java

```diff
@@ -0,0 +1,115 @@
+        assertEquals(1, courseList.size());
+        assertEquals(instructor2OfCourse3.courseId, courseList.get(0).getId());
+        assertTrue(CoursesLogic.inst().isCoursePresent(instructor2OfCourse3.courseId));
+
+        ______TS("Not enough parameters");
+        verifyAssumptionFailure();
+
+        ______TS("Typical case, delete 1 course from Recycle Bin, with privilege");
+        InstructorAttributes instructor1OfCourse3 = typicalBundle.instructors.get("instructor1OfCourse3");
+        String instructor1Id = instructor1OfCourse3.googleId;
+
+        gaeSimulation.loginAsInstructor(instructor1Id);
+
+        submissionParams = new String[] {
+                Const.ParamsNames.COURSE_ID, instructor1OfCourse3.courseId
+        };
+
+        assertTrue(CoursesLogic.inst().isCoursePresent(instructor1OfCourse3.courseId));
+        deleteAction = getAction(submissionParams);
+        RedirectResult redirectResult = getRedirectResult(deleteAction);
+
+        assertEquals(
+                getPageResultDestination(Const.ActionURIs.INSTRUCTOR_COURSES_PAGE, false, "idOfInstructor1OfCourse3"),
+                redirectResult.getDestinationWithParams());
+        assertFalse(redirectResult.isError);
+        assertEquals("The course idOfTypicalCourse3 has been permanently deleted.",
+                redirectResult.getStatusMessage());
+
+        assertFalse(CoursesLogic.inst().isCoursePresent(instructor1OfCourse3.courseId));
+
+        String expectedLogMessage = "TEAMMATESLOG|||instructorRecoveryDeleteCourse|||instructorRecoveryDeleteCourse|||"
+                + "true|||Instructor|||Instructor 1 of Course 3|||idOfInstructor1OfCourse3|||"
+                + "instr1@course3.tmt|||Course deleted: idOfTypicalCourse3|||"
+                + "/page/instructorRecoveryDeleteCourse";
+        AssertHelper.assertLogMessageEquals(expectedLogMessage, deleteAction.getLogMessage());
+
+    }
+
+    @Override
+    protected InstructorCourseDeleteRecoveryCourseAction getAction(String... params) {
+        return (InstructorCourseDeleteRecoveryCourseAction) gaeSimulation.getActionObject(getActionUri(), params);
+    }
+
+    @Override
+    @Test
+    protected void testAccessControl() throws Exception {
+        CoursesLogic.inst().createCourseAndInstructor(
+                typicalBundle.instructors.get("instructor1OfCourse3").googleId,
+                "icdat.owncourse", "New course", "UTC");
+
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.COURSE_ID, "icdat.owncourse"
+        };
+
+        //  Test access for users
+        verifyUnaccessibleWithoutLogin(submissionParams);
+        verifyUnaccessibleForUnregisteredUsers(submissionParams);
+        verifyUnaccessibleForInstructorsOfOtherCourses(submissionParams);
+        verifyUnaccessibleForStudentsOfOtherCourses(submissionParams);
+        verifyUnaccessibleWithoutModifyCoursePrivilege(submissionParams);
+    }
```

Comment:
Oh this is because in `BaseActionTest.java` it is all using the original instructors & courses in `typicalDataBundle.json` (i.e. `instructor1`, `helper`, `course1`) for tests. If I change it then all the other action tests will fail. So actually implementing `testAccessControl()` method here is just to make this action class extend `BaseActionTest`. I've added more tests in `testExecuteAndPostProcess()` to check accessibility of other instructors of the same course, compared to other action tests.